tests/main/interfaces-network-control/task.yaml

```diff
@@ -76,7 +76,7 @@ execute: |
             echo "Expected error caling command with disconnected plug"
             exit 1
         fi
-        MATCH "Permission denied" < net-query.output
+        MATCH "(Permission denied|Operation not permitted)" < net-query.output
```

Comment:
only after manually changing the seccom profile to `@complain` I was able to see this denial from apparmor:
```
Mar 06 12:15:32 mar061134-824633.c.snapd-spread.internal kernel: audit: type=1400 audit(1709727332.849:190): apparmor="DENIED" operation="capable" class="cap" profile="snap.network-control-consumer.cmd" pid=34335 comm="python3" capability=13  capname="net_raw"
```
so clearly this got blocked by seccomp. There's other question, the logging is supposedly supported (as I changed the snap-seccomp code to print in case it is), but how do I see the action being logged?

Reply:
Perhaps it was lost due to ratelimiting or sth:
```
Mar 06 12:28:19 mar061134-824633.c.snapd-spread.internal kernel: audit: type=1300 audit(1709728099.227:348): arch=c000003e syscall=257 success=no exit=-13 a0=ffffff9c a1=79ef59e2e9d0 a2=90800 a3=0 items=0 ppid=32886 pid=34898 auid=0 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=pts0 ses=1 comm="python3" exe="/usr/bin/python3.8" subj=snap.network-control-consumer.cmd>
Mar 06 12:28:19 mar061134-824633.c.snapd-spread.internal kernel: audit: type=1327 audit(1709728099.227:348): proctitle=707974686F6E33002D6300696D706F727420736F636B65743B2041465F5844503D34343B20736F636B65742E736F636B65742841465F5844502C20736F636B65742E534F434B5F52415729
```

Reply:
Perhaps the version of libseccomp or the kernel used in noble supports newer seccomp restrictions (or maybe glibc / python in noble has a change to implement an existing system call which then requires an additional permission) - and since our seccomp policy is an allow-list then we are likely missing some permission that is now being denied? We can get some additional info from the denials listed above:

```
$ scmp_sys_resolver 257
openat

$ errno 13
EACCES 13 Permission denied

$ aa-decode \
707974686F6E33002D6300696D706F727420736F636B65743B2041465F5844503D34343B20736F636B65742E736F636B65742841465F5844502C20736F636B65742E534F434B5F52415729
Decoded: python3-cimport socket; AF_XDP=44; socket.socket(AF_XDP, socket.SOCK_RAW)
```

So we are seemingly failing to create an AF_XDP socket in RAW mode when executing something that seemingly its cmd set as `python3-cimport socket; AF_XDP=44; socket.socket(AF_XDP, socket.SOCK_RAW)` - but I have no idea what `python3-cimport` is...

Reply:
I think arguments are just separated with \x00:
```
In [5]: bytes.fromhex("707974686F6E33002D6300696D706F727420736F636B65743B2041465F5844503D34343B20736F636B65742E736F636B65742841465F5844502C20736F636B65742E534F434B5F52415729")
Out[5]: b'python3\x00-c\x00import socket; AF_XDP=44; socket.socket(AF_XDP, socket.SOCK_RAW)'
```

Reply:
@alexmurray just to be sure, are we certain that the apparmor socket mediation works and we're not just papering over a potential issue? This is just one of the network related tests that fail on 24.04, the others are:
google:ubuntu-24.04-64:tests/main/interfaces-network
google:ubuntu-24.04-64:tests/main/interfaces-network-bind
In both cases resources should not be accessible but are.